eventsourcing/src/main/java/org/axonframework/eventsourcing/GenericAggregateFactory.java

```diff
@@ -80,17 +84,21 @@ public GenericAggregateFactory(AggregateModel<T> aggregateModel) {
     @SuppressWarnings({"unchecked"})
     @Override
     protected T doCreateAggregate(String aggregateIdentifier, DomainEventMessage firstEvent) {
-        return aggregateModel.type(firstEvent.getType())
-                             .map(this::newInstance)
-                             .orElseThrow(() -> new IncompatibleAggregateException(format(
-                                     "The [%s] aggregate does not exist.",
-                                     firstEvent.getType())));
+        return aggregateModel().type(firstEvent.getType())
+                               .map(this::newInstance)
+                               // backwards compatibility in cases firstEvent does not contain the aggregate type
+                               .orElseGet(() -> aggregateModel().defaultConstructor(getAggregateType())
```

Comment:
There seem to be 2 different "levels of expectation" in the API. The model returns a Type based on the type from the event, and if that is not found, it asks for the default constructor from the AggregateModel. The latter seems a bit out of place: the Model assumes default constructors are used to instantiate empty aggregates. That assumption belongs in the GenericAggregateFactory.
I suggest something along the lines of:
```
newInstance(aggregateModel().type(firstEvent.getType())
                             .orElse(getAggregateType));
```
In that case, AggregateModel doesn't need to expose any default constructors. Just types.

Reply:
I moved inspection of constructors to the model to be inline with https://github.com/AxonFramework/AxonFramework/pull/1317#discussion_r369149500 . Where do you propose to inspect constructors?

Reply:
Addressed.